flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CoordinatorShutdownTest.java

```diff
@@ -83,6 +86,10 @@ public void testCoordinatorShutsDownOnFailure() {
 						CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION,
 						true),
 					null));
+
+			ExecutionConfig executionConfig = new ExecutionConfig();
+			executionConfig.setRestartStrategy(RestartStrategies.fallBackRestart());
```

Comment:
Should we maybe set the `FallbackRestartStrategyConfiguration` per default in the `ExecutionConfig`? That way, we could also simplify the resolve code.

Reply:
You're right. I'm just wondering whether you ever want to enable checkpointing without a restart strategy. So to speak if you set `FallbackRestartStrategy`, enable checkpointing and set `NoRestartStrategy` as the server side `RestartStrategy`, then do you want `FixedRestartStrategy` or `NoRestartStrategy`?

On the other hand you might want to disable restarting for all jobs running on your cluster by setting the restart strategy to `NoRestartStrategy`.

Maybe the proper solution would be to set `ExecutionConfig#restartStrategy` to `FallbackRestartStrategy` and introduce a new default server side restart strategy `NoOrFixedIfCheckpointingEnabled` which resolved to `FixedRestartStrategy` if checkpointing is enabled and if not it resolves to `NoRestartStrategy`.

What do you think?